base/applications/sdbinst/sdbinst.c

```diff
@@ -0,0 +1,957 @@
+/*
+ * PROJECT:     ReactOS sdbinst
+ * LICENSE:     GPL-2.0-or-later (https://spdx.org/licenses/GPL-2.0-or-later)
+ * PURPOSE:     Application compatibility database installer
+ * COPYRIGHT:   Copyright 2020 Max Korostil (mrmks04@yandex.ru)
+ */
+
+
+#include <windef.h>
+#include <winbase.h>
+#include <tchar.h>
+#include <winreg.h>
+#include <strsafe.h>
+#include <objbase.h>
+#include <apphelp.h>
+
+
+
+#define APPCOMPAT_CUSTOM_REG_PATH L"Software\\Microsoft\\Windows NT\\CurrentVersion\\AppCompatFlags\\Custom"
+#define APPCOMPAT_LAYERS_REG_PATH L"Software\\Microsoft\\Windows NT\\CurrentVersion\\AppCompatFlags\\Layers"
+#define APPCOMPAT_INSTALLEDSDB_REG_PATH L"Software\\Microsoft\\Windows NT\\CurrentVersion\\AppCompatFlags\\InstalledSDB"
+#define UNINSTALL_REG_PATH L"Software\\Microsoft\\Windows\\CurrentVersion\\Uninstall"
+#define DBPATH_KEY_NAME L"DatabasePath"
+
+HRESULT
+RegisterSdbEntry(
+    _In_ PWCHAR sdbEntryName,
+    _In_ LPCWSTR dbGuid,
+    _In_ ULONGLONG time,
+    _In_ BOOL isInstall,
+    _In_ BOOL isExe)
+{
+    PWCHAR regName;
+    HKEY  hKey = NULL;
+    LSTATUS status;
+    HRESULT hres;
+
+    regName = (PWCHAR)HeapAlloc(GetProcessHeap(), 0, MAX_PATH * sizeof(WCHAR));
+    if (!regName)
+    {
+        hres = HRESULT_FROM_WIN32(ERROR_NOT_ENOUGH_MEMORY);
+        goto end;
+    }
+    ZeroMemory(regName, MAX_PATH * sizeof(WCHAR));
+
+    hres = StringCchPrintf(regName, MAX_PATH, L"%ls\\%ls",
+                     isExe ? APPCOMPAT_CUSTOM_REG_PATH : APPCOMPAT_LAYERS_REG_PATH, sdbEntryName);
```

Comment:
```suggestion
    hres = StringCchPrintfW(regName, MAX_PATH, L"%ls\\%ls",
                            isExe ? APPCOMPAT_CUSTOM_REG_PATH : APPCOMPAT_LAYERS_REG_PATH,
                            sdbEntryName);
```
Also if you use the fixed size buffer for regName (without the HeapAlloc call), the MAX_PATH here could be converted into `ARRAYSIZE(regSize)`.